### PR TITLE
Refactor header sorting logic to be more clear

### PR DESCRIPTION
**Pull request by scramsby**

### CODE REVIEW

File: iwyu_output.cc [hunk 1]
```diff
@@ -1774,29 +1774,45 @@ OutputLine PrintableIncludeOrForwardDeclareLine(
...
+
+LineSortOrdinal GetLineSortOrdinal(const OneIncludeOrForwardDeclareLine& line,
+                                   const set<string>& associated_quoted_includes,
+                                   const IwyuFileInfo* file_info) {
+  if (!line.IsIncludeLine())
+    return LineSortOrdinal::kForwardDeclaration;
+  if (file_info && file_info->is_pch_in_code())
+    return LineSortOrdinal::kPrecompiledHeader;
+
+  const auto& quotedInclude = line.quoted_include();
```
Comment (kimgr) [inline iwyu_output.cc]: No `auto` if the type is not visible on the rhs of the assignment, please. Except in emergency cases like map iteration :)

Comment (kimgr) [reply iwyu_output.cc]: And I'd prefer the concrete type to be `std::string` (sans reference) -- I find lifetime extension to be a wart on the language (among many)

File: iwyu_output.cc [hunk 2]
```diff
@@ -1774,29 +1774,45 @@ OutputLine PrintableIncludeOrForwardDeclareLine(
     GetSymbolsSortedByFrequency(line.symbol_counts()));
 }
 
-typedef pair<int, string> LineSortKey;
+enum class LineSortOrdinal {
+  kPrecompiledHeader,
```
Comment (kimgr) [inline iwyu_output.cc]: Hmm, I'm also unsure about the naming of enum class enumerators. I think the `k` prefix has outlived its usefulness here, as the names are scoped in `LineSortOrdinal` anyway. I vote to skip the k-prefix.

File: iwyu_output.cc (hunk 1 above)
Comment (scramsby) [reply iwyu_output.cc]: Sure thing.

File: iwyu_output.cc (hunk 2 above)
Comment (scramsby) [reply iwyu_output.cc]: I totally agree. Was just trying to follow existing convention. I'll update and squash.

**Comment by kimgr** [comment]
Thanks, looks good to me!

**Pull request merged**